<a href="https://colab.research.google.com/github/sovikapoudel/Data-Visualiation-Spring-2022/blob/main/Quiz_6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<h1><center>Data Visualization Spring 2022</center></h1>

---

###Assignment: 5
Name: Sovika Poudel <br>
Year: Junior <br>
Course: CPSMA-4313 Data Processing Visualization 
### For this part of Assignment I am going to load the libraries, gather dataset and load them here in this notebook and copy all my work from previous assignment for the better and easier reference.

In [ ]:
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

Authenticated


Gathering Data

In [ ]:
%%bigquery --project data-vis-assignment-338801
SELECT *
  From `bigquery-public-data.austin_bikeshare.bikeshare_trips`
  lIMIT 5



trip_id  ... duration_minutes
0  9900289692  ...               39
1  9900285987  ...               31
2  9900285989  ...               31
3  9900285991  ...               30
4  9900286140  ...               19

[5 rows x 9 columns]

What was the second most popular starting station?

In [ ]:
%%bigquery --project data-vis-assignment-338801 
SELECT start_station_name, COUNT(start_station_id) as total_starting_point
  From `bigquery-public-data.austin_bikeshare.bikeshare_trips`
  GROUP BY start_station_name
  Order BY total_starting_point DESC ###this will give the most popular on the top so we can see 2nd most popular in 2nd row
  LIMIT 5

start_station_name  total_starting_point
0      21st & Speedway @PCL                 72799
1      Riverside @ S. Lamar                 40635
2  City Hall / Lavaca & 2nd                 36520
3            2nd & Congress                 35307
4      Rainey St @ Cummings                 34758

So the second most popular starting station is Riverside @ S. Lamar	with 40635 counts

How many trips lasted over an hour and were a round trip (started and stopped at the same station)?

At first, for this part I am just showing the list in order and confirming if the query is running correctly.

In [ ]:
%%bigquery --project data-vis-assignment-338801 
SELECT start_station_name, end_station_name, duration_minutes as time_travelled
  From `bigquery-public-data.austin_bikeshare.bikeshare_trips`
  WHERE  start_station_name = end_station_name and  duration_minutes >60
  LIMIT 100

start_station_name  ... time_travelled
0                   Toomey Rd @ South Lamar  ...             82
1                          Waller & 6th St.  ...             76
2                          Waller & 6th St.  ...             75
3                          Waller & 6th St.  ...             74
4                          Waller & 6th St.  ...             73
..                                      ...  ...            ...
95                  Toomey Rd @ South Lamar  ...             88
96                  Toomey Rd @ South Lamar  ...             73
97                  Toomey Rd @ South Lamar  ...             72
98                         Waller & 6th St.  ...             61
99  Capital Metro HQ - East 5th at Broadway  ...            192

[100 rows x 3 columns]

Now, for the total numbers of loop that were over 60 minutes I am using a clause within a 'WHERE' clause

In [ ]:
%%bigquery --project data-vis-assignment-338801 
SELECT COUNT(*) as No_of_Overthehour_loops
  From `bigquery-public-data.austin_bikeshare.bikeshare_trips`
  WHERE  start_station_name = end_station_name and  duration_minutes >60
  
  # Order by time_travelled DESC ### this line of code is not working correctly on this part. so commenting this out for now.
 

No_of_Overthehour_loops
0                    54795

##Assignment 6 begins from here


###Task to do!

Examine the average trip on Austin bike share grouping by property_type column from the bikeshare_stations table. Is average time on trips starting from 'parkland' longer than from 'sidewalks'? To answer this question you must join the bikeshare_trips and bikeshare_stations on the start_station_name and the name from their respective tables.

SQL Joins

In [37]:
%%bigquery --project data-vis-assignment-338801 
SELECT *
FROM `bigquery-public-data.austin_bikeshare.bikeshare_stations`
Limit 10

station_id  ...             modified_date
0        3464  ... 2021-01-04 12:00:00+00:00
1        2500  ... 2021-01-04 12:00:00+00:00
2        2536  ... 2021-01-04 12:00:00+00:00
3        2538  ... 2021-01-04 12:00:00+00:00
4        2541  ... 2021-01-04 12:00:00+00:00
5        2545  ... 2021-01-04 12:00:00+00:00
6        2546  ... 2021-01-04 12:00:00+00:00
7        2550  ... 2021-01-04 12:00:00+00:00
8        2564  ... 2021-01-04 12:00:00+00:00
9        2576  ... 2021-01-04 12:00:00+00:00

[10 rows x 14 columns]

First, let's gather the data about the property type and it's count.

In [ ]:
%%bigquery --project data-vis-assignment-338801 
SELECT property_type as Type, COUNT(property_type ) as Num
FROM `bigquery-public-data.austin_bikeshare.bikeshare_stations` 
WHERE property_type != " " ##There's one blank entry in the dataset under property_type so I am using the where clause to remove that from my output
group by  property_type
LIMIT 20 #just checking if this pulls out the all the informations, we only have total of 6 entries for this query though

Type  Num
0              parkland    6
1              sidewalk   28
2          paid_parking   19
3    nonmetered_parking    8
4  undetermined_parking   19

Now, we have got the total numbers of each property types. Let's join the datasets for our next part.

In [ ]:
%%bigquery --project data-vis-assignment-338801 
WITH t as(
SELECT *
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
WHERE start_station_name = "Waller & 6th St.")

SELECT table1.bikeid, SUM(CASE WHEN table1.duration_minutes < table2.duration_minutes THEN 1 ELSE 0 END) as Duration_minutes_difference
FROM t as table1 JOIN t as table2 ON table1.start_station_name = table2.start_station_name
GROUP BY table1.bikeid


bikeid  Duration_minutes_difference
0      869                        45841
1      117                        39552
2      965                        57903
3      983                        29175
4      198                        25767
..     ...                          ...
396    340                         8252
397    507                        19303
398    873                        10239
399    233                         2937
400    851                         1193

[401 rows x 2 columns]

In [ ]:
%%bigquery --project data-vis-assignment-338801
SELECT *
  From `bigquery-public-data.austin_bikeshare.bikeshare_stations`
  lIMIT 5

station_id  ...             modified_date
0        3464  ... 2021-01-04 12:00:00+00:00
1        2500  ... 2021-01-04 12:00:00+00:00
2        2536  ... 2021-01-04 12:00:00+00:00
3        2538  ... 2021-01-04 12:00:00+00:00
4        2541  ... 2021-01-04 12:00:00+00:00

[5 rows x 14 columns]

In [ ]:
%%bigquery --project data-vis-assignment-338801 
WITH t as(
SELECT *
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
WHERE start_station_name = "Waller & 6th St.")

SELECT *
FROM t as table1 JOIN t as table2 ON table1.start_station_name = table2.start_station_name
Limit 10

trip_id  ... duration_minutes_1
0  9900286171  ...                  6
1  9900286540  ...                  6
2  9900287861  ...                  6
3  9900287864  ...                  6
4  9900288274  ...                  6
5  9900288276  ...                  6
6  9900288278  ...                  6
7  9900288279  ...                  6
8  9900288445  ...                  6
9  9900288447  ...                  6

[10 rows x 18 columns]